scripts/sysman/firmware.yml

```diff
@@ -90,6 +90,49 @@ returns:
     - $X_RESULT_ERROR_INSUFFICIENT_PERMISSIONS:
         - "User does not have permissions to perform this operation."
 --- #--------------------------------------------------------------------------
+type: function
+desc: "Find Firmware Flash Progress"
+version: "1.7"
+class: $sFirmware
+name: UpdateProgress
+details:
+    - "The application may call this function from simultaneous threads."
+    - "The implementation of this function should be lock-free."
+params:
+    - type: $s_firmware_handle_t
+      name: hFirmware
+      desc: "[in] Handle for the component."
+    - type: "uint32_t*"
+      name: pCompletionPercent
+      desc: "[in,out] Pointer to the Completion Percentage of Firmware Update"
+returns:
+    - $X_RESULT_ERROR_INSUFFICIENT_PERMISSIONS:
+        - "User does not have permissions to perform this operation"
```

Comment:
does this require sudo permissions? if not, do we need this return?

Reply:
Firmware update need sudo permissions

Reply:
@vaidyan1-int but this is a getter right? not for fw update, but just to gather progress?

Reply:
i agree with @aravindksg  firmwareupdate progress just showing percentage of firmware update done